packages/react-native-renderer/src/ReactNativePublicCompat.js

```diff
@@ -218,3 +224,44 @@ export function getNodeFromInternalInstanceHandle(
     internalInstanceHandle.stateNode.node
   );
 }
+
+// Remove this once Paper is no longer supported and DOM Node API are enabled by default in RN.
+export function isChildPublicInstance(
+  parentInstance: FabricPublicInstance | PaperPublicInstance,
+  childInstance: FabricPublicInstance | PaperPublicInstance,
+): boolean {
+  if (__DEV__) {
+    const parentInternalInstanceHandle =
+      // $FlowExpectedError[incompatible-call] Prioritizing early return for Fabric case, without explicitly checking with instanceof to make Flow happy.
```

Comment:
Would this be necessary if you flipped the logic to check Paper first via `instanceof`?